backend/canisters/community/api/src/updates/report_message.rs

```diff
@@ -9,15 +9,19 @@ pub struct Args {
     pub message_id: MessageId,
     pub reason_code: u32,
     pub notes: Option<String>,
+    pub delete: bool,
 }
 
 #[derive(CandidType, Serialize, Deserialize, Debug)]
 pub enum Response {
     Success,
     CommunityFrozen,
     UserNotInCommunity,
+    UserSuspended,
     ChannelNotFound,
+    NotAuthorized,
     UserNotInChannel,
     MessageNotFound,
+    MessageTypeCannotBeDeleted,
```

Comment:
We should probably support deleting all messages. Otherwise anyone can make their message permanent by sending 0.0000001 ICP